src/Container/LCParticleContainer.cpp

```diff
@@ -222,15 +284,21 @@ std::vector<Particle *> LCParticleContainer::getBoundaryParticles() {
   return result;
 }
 
+
 void LCParticleContainer::handleBoundaryAction() {
   std::vector<Particle*> boundaryparticles = getBoundaryParticles();
+  std::array<double, 12> bounds;
     for(auto p: boundaryparticles) {
-      std::array<double, 6> bounds = getInfluencingBoundarysWithDistance(p);
+      bounds = getInfluencingBoundarysWithDistance(p);
         for(int i = 0; i < 6; i++) {
-          if(boundary_types[i] == 1){} //outflow
+          if(boundary_types[i] == 1) {
+            if(bounds.at(i) < 0) {
+              p->park();
+            }
+          } //outflow
           else if(boundary_types[i] == 2){
             // generate halo Particle
-            if(bounds.at(i) != -1.0 && abs(bounds.at(i)) <= r_cutoff){ //&& abs(bounds.at(i)) <= r_cutoff
+            if(bounds.at(i) != -1.0 && abs(bounds.at(i)) <= pow(2, 1/6)){ //&& abs(bounds.at(i)) <= r_cutoff
               if( i == 0 ){ //Boundary to YZ Plane
                 std::array<double, 3> x_arg = {-bounds.at(i), p->getX().at(1), p->getX().at(2)};
                 std::array<double, 3> v_arg = {-1*(p->getV().at(0)), p->getV().at(1), p->getV().at(2)};
```

Comment:
You just need the position (not velocity) to calculate the force in handling the boundary condition, so you don't need to worry about the velocity (you can save on all the calculations in this function)